### ISSUE TITLE
R^2 is not accuracy

### ISSUE THREAD

**Issue by desertnaut** [S-B-Iqbal/Predicting-Power-Output-of-a-combined-cycle-power-plant.#1]
Please notice that your claim in README that

> The model is able to achieve an **accuracy of 93%** on Test Data

is wrong. What you calculate is the R^2, which has nothing to do with accuracy. Moreover, in ML contexts, accuracy is a term reserved for *classification* problems, and not for regression ones, such as the one you solve here. See the discussion in the Stack Overflow thread[ What function defines accuracy in Keras when the loss is mean squared error (MSE)?](https://stackoverflow.com/questions/48775305/what-function-defines-accuracy-in-keras-when-the-loss-is-mean-squared-error-mse) for details.

**Comment by S-B-Iqbal** [comment]
@desertnaut Thanks for the comment.

- Firstly, Kindly refer section **3.1.3** of [**_An Introduction to Statistical Learning_**](http://faculty.marshall.usc.edu/gareth-james/ISL/). Hope this helps in fortifying your understanding of how accuracy is measured in Regression models.

- Secondly, In raising future issues, please make sure you provide reference of standard peer reviewed Literature. It helps the whole DS Community to learn more. Hope that helps!

Regards,
S B Iqbal
Romans 1:22 **Professing themselves to be wise, they became fools.**

**Comment by desertnaut** [comment]
Thanks for the reply, however unkind it may be.

Please notice the following:

- I kindly suggest you go a little beyond the header of ISL Section 3.1.3; that way, you may discover that the term "accuracy" is never mentioned again in the section (effectively replaced with the terms "quality" and "measure" of the fit), and with good reason. 

- Especially with the R^2 statistic (which the comment was specifically about), you will not find it mentioned in the same sentence with the term "accuracy" anywhere in the whole ISL book, again with good reason: R^2 is used there as a goodness-of-fit measure (i.e. with the *training* data), while the term "accuracy", when used in its everyday-life meaning (it does indeed, in places) is almost always used in a *prediction* setting (with *unseen* data). Kindly refer to section 2.2 of ISL where again, apart from a single reference in p. 30, the term "accuracy" appears nowhere else except in the title.

- Similar is the treatment of the term "accuracy" in another DS "bible", [Applied Predictive Modeling](http://appliedpredictivemodeling.com/), where, apart again from its use in the everyday meaning in the first (general) chapters of the book, every subsequent single occurrence of the term is always in a classification context, and never in a regression one.

- The real issue here is to discriminate between the everyday-life use of the term "accuracy" and the one used in an ML setting: since the term is used to declare a very specific measure in classification settings, it is not advised to use it in such ML settings while implying its everyday use (which is actually what you do here, and what the comment was about). The practitioner community (sorry, it exists along the academic one, and everyone is free to ignore this fact at their own peril) is adamantly clear that the term "accuracy" is meant to be [classification accuracy](https://developers.google.com/machine-learning/crash-course/classification/accuracy), and not *anything* else. Peer-reviewed or not, you will not find a single use of the term "accuracy" in regression settings in the relevant page of the scikit-learn [documentation](https://scikit-learn.org/stable/modules/model_evaluation.html), again with good reason (if you believe they are in the wrong, I guess you are always welcome to open an issue at their own Github repo).

- Your insistence on using only peer-reviewed literature is unsurprising but unfounded and ill-guided (contrary to the hype some want the world to believe, data science *practitioners* do not sleep with research articles under their pillow, and their learning material is actually much wider). If you had gone beyond the surface, e.g. in the more advanced textbook [The Elements of Statistical Learning](https://web.stanford.edu/~hastie/ElemStatLearn/) (it shares authorship with the more introductory ISL), you would have read Hastie et al., while explaining a (huge...) mistake in the cross-validation procedure when combined with feature selection, stating that:

> While this point may seem obvious to the reader, we have seen this **blunder** committed many times in **published papers in top rank journals**.

(emphasis mine)

- And however deeply we may respect giants in the field like Hastie and Tibshirani, their insistence in section 3.1.3 of ISL (you know, the same one you sent me to read so that I fortify my understanding...) that R^2 is alsways between 0 and 1 is **just wrong**, since a model can in principle be arbitrarily *bad*, hence leading to a negative R^2. Don't believe me? Well, you may require a peer-reviewed article, or simply [**do an experiment**](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.r2_score.html).

The point of the comment was simply that, if you want to state an R^2 of 93%, then you should simply say "*an R^2 of 93%*", and that should be all. No need to undermine the concept with unnecessary references to the "accuracy". Your concern to help "*the whole DS Community to learn more*" is touching but unconvincing: it's because of such unfortunate references read by poor people trying to learn that me (and other poor people like me at Stack Overflow) have to spend our precious time trying to [undo the damage](https://stackoverflow.com/questions/48775305/what-function-defines-accuracy-in-keras-when-the-loss-is-mean-squared-error-mse) done by such misinformed "learning" material. And since you have explicitly expressed a "*Hope that helps!*" in your reply, let me be equally explicit here: **it did not**.

Let me be crystal clear that your tone (and smart-ass quotes of Romans 1:22) is certainly **not appreciated**.
